Applications/Spire/Source/Ui/QuantityBox.cpp

```diff
@@ -34,4 +35,50 @@ QuantityBox::QuantityBox(std::shared_ptr<OptionalQuantityModel> model,
   update_style(get_decimal_box(), [&] (auto& style) {
     style.get(Any()).set(TrailingZeros(0));
   });
+  get_decimal_box().installEventFilter(this);
+}
+
+bool QuantityBox::eventFilter(QObject* watched, QEvent* event) {
+  if(watched == &get_decimal_box() && event->type() == QEvent::Show) {
+    auto proxy = find_focus_proxy(get_decimal_box());
+    if(proxy) {
+      proxy->installEventFilter(this);
+    }
+  } else if(event->type() == QEvent::KeyPress && !is_read_only()) {
+    auto& key_event = *static_cast<QKeyEvent*>(event);
```

Comment:
Why is an event filter used instead of having the ```QKeyEvent``` bubble up?

Reply:
Since the `QKeyEvent` is set to accepted state by the `QLineEdit` contained in the `TextBox` which is used by the `DecimalBox` when the `QLineEdit` receives alphanumeric input, it can't be propagated to its parent widget according to the description of `QKeyEvent`:

> Calling [ignore](https://doc.qt.io/qt-5/qevent.html#ignore)() on a key event will propagate it to the parent widget. The event is propagated up the parent widget chain until a widget accepts it or an event filter consumes it.

Reply:
Sounds good.